app/frontend/components/domains/permit-application/revision-sidebar.tsx

```diff
@@ -0,0 +1,248 @@
+import { useMountStatus } from "../../../hooks/use-mount-status"
+import { IPermitApplication } from "../../../models/permit-application"
+import { IRevisionRequestsAttributes } from "../../../types/api-request"
+import { IFormIORequirement, IRevisionRequest } from "../../../types/types"
+import { getRequirementByKey } from "../../../utils/formio-component-traversal"
+import { getSinglePreviousSubmissionJson } from "../../../utils/formio-submission-traversal"
+import { handleScrollToBottom } from "../../../utils/utility-functions"
+import ConfirmationModal from "../../shared/modals/confirmation-modal"
+import { ScrollLink } from "../../shared/permit-applications/scroll-link"
+import { RevisionModal } from "../../shared/revisions/revision-modal"
+
+interface IRevisionSideBarProps {
+  permitApplication: IPermitApplication
+  onCancel?: () => void
+  sendRevisionContainerRef?: MutableRefObject<HTMLDivElement>
+  forSubmitter?: boolean
+}
+
+export interface IRevisionRequestForm {
+  revisionRequestsAttributes: IRevisionRequestsAttributes[]
+}
+
+export const RevisionSideBar = observer(
+  ({ permitApplication, onCancel, sendRevisionContainerRef, forSubmitter }: IRevisionSideBarProps) => {
+    const { t } = useTranslation()
+    const isMounted = useMountStatus()
+    const [requirementForRevision, setRequirementForRevision] = useState<IFormIORequirement>()
+    const [submissionJsonForRevision, setSubmissionJsonForRevision] = useState<any>()
+    const [revisionRequest, setRevisionRequest] = useState<IRevisionRequest>()
+    const navigate = useNavigate()
+
+    const getDefaultRevisionRequestValues = () => ({
+      revisionRequestsAttributes: permitApplication.revisionRequests as IRevisionRequestsAttributes[],
+    })
+
+    const revisionFormMethods = useForm<IRevisionRequestForm>({
+      defaultValues: getDefaultRevisionRequestValues(),
+    })
+
+    const { handleSubmit, control, reset } = revisionFormMethods
+
+    const useFieldArrayMethods = useFieldArray({
+      control,
+      name: "revisionRequestsAttributes",
+      keyName: "fieldId",
+    })
+
+    const { fields } = useFieldArrayMethods
+
+    useEffect(() => {
+      reset(getDefaultRevisionRequestValues())
+    }, [permitApplication?.revisionRequests?.length])
+
+    const onSaveRevision = (formData: IRevisionRequestForm) => {
+      permitApplication.updateRevisionRequests(formData)
+    }
+
+    const [topHeight, setTopHeight] = useState<number>()
+
+    useEffect(() => {
+      const permitHeaderHeight = document.getElementById("permitHeader")?.offsetHeight
+      setTopHeight(permitHeaderHeight)
+    }, [isMounted, window.innerWidth, window.innerHeight])
+
+    const { isOpen, onOpen, onClose } = useDisclosure()
+
+    const onFinalizeRevisions = () => {
+      ;(async () => {
+        const ok = await permitApplication.finalizeRevisionRequests()
+        if (ok) navigate(`/jurisdictions/${permitApplication.jurisdiction.slug}/submission-inbox`)
+      })()
```

Comment:
what's the point of the IIFE? Just make the top level function async do the same thing in the function body